### PR TITLE
Initial round of revisions

### PR DESCRIPTION
**Pull request by alicederyn**
Revisions proposed by @dmoissett

### CODE REVIEW

File: pep-9999.rst [hunk 1]
```diff
@@ -36,7 +35,7 @@ For example, given the following definition::
         else:
             return f'{movie["name"]} ({movie["year"]})'
 
-A type checker should accept this code::
+A type checker should then consider this code type safe::
```
Comment (alicederyn) [inline pep-9999.rst:38]: I took this language straight from https://peps.python.org/pep-0589/ -- happy to change if you still prefer it

File: pep-9999.rst [hunk 2]
```diff
@@ -90,15 +97,19 @@ This could be avoided by using a :pep:`544` protocol::
         @overload
         def __getitem__(self, key: Literal["year"]) -> int | None: ...
 
-This is very repetitive, easy to get wrong, and is still missing important definitions like ``in`` and ``keys()``.
+This is very repetitive, easy to get wrong, and is still missing important method definitions like ``__contains__()`` and ``keys()``.
+
+The proposed ``TypedMapping`` type allows a straightforward way of defining these types that should be familiar to existing users of ``TypedDict`` and support the cases exemplified above.
+In addition to those benefits, by flagging arguments of a function as ``TypedMapping`` makes explicit not just to typecheckers but also to user that the function is not going to modify its argument, which is usually a desirable property of a function interface.
```
Comment (alicederyn) [inline pep-9999.rst]:
```suggestion
In addition to those benefits, by flagging arguments of a function as ``TypedMapping``, it makes explicit not just to typecheckers but also to users that the function is not going to modify its inputs, which is usually a desirable property of a function interface.
```

File: pep-9999.rst [hunk 3]
```diff
@@ -90,15 +97,19 @@ This could be avoided by using a :pep:`544` protocol::
...
+Finally, this allows bringing the benefits of ``TypedDict``to other mapping types that are unrelated to ``dict``.
```
Comment (alicederyn) [inline pep-9999.rst]:
```suggestion
Finally, this allows bringing the benefits of ``TypedDict`` to other mapping types that are unrelated to ``dict``.
```

File: pep-9999.rst [hunk 4]
```diff
@@ -90,15 +97,19 @@ This could be avoided by using a :pep:`544` protocol::
...
 
 Specification
 =============
 
-The ``TypedMapping`` type is a protocol behaving identically to ``TypedDict``, except:
+The ``TypedMapping`` type is a protocol behaving almost identically to ``TypedDict``, except:
 
-1. instances need not be subclasses of dict
-2. no mutate methods will be generated
+1. Its instances need not be instances of subclasses of ``dict``
```
Comment (alicederyn) [inline pep-9999.rst]:
```suggestion
1. The runtime type is not constrained to be a ``dict``
```
Realised we can simplify this as, according to https://peps.python.org/pep-0589/:
> the runtime type of a TypedDict object will always be just dict (it is never a subclass of dict)

File: pep-9999.rst [hunk 5]
```diff
@@ -127,16 +138,17 @@ For example::
...
      
-    movie["year"] = 1985  # Fine, mutate methods added in definition
-    movie["name"] = "Terminator"  # Error, name mutator not declared
+    movie["year"] = 1985  # Fine, mutator methods added in definition
+    movie["name"] = "Terminator"  # Type check error, "name" mutator not declared
 
 
-Type Consistency
-----------------
+Type Consistency Rules
```
Comment (alicederyn) [inline pep-9999.rst:145]: Copied this heading from https://peps.python.org/pep-0589/ too but happy to change it :)

File: pep-9999.rst [hunk 6]
```diff
@@ -48,32 +47,40 @@ A type checker should accept this code::
...
             return f'{movie["name"]} ({movie["year"]})'
 
-Attempting to pass a ``MovieRecord`` into this function gives (in mypy) the error::
+Attempting to pass a ``MovieRecord`` into ``movie_string`` results (using mypy) in the error::
 
     Argument 1 to "movie_string" has incompatible type "MovieRecord"; expected "Movie"
 
-This incompatibility is due to the unused mutate methods generated on ``Movie``.
-This could be avoided by using a :pep:`544` protocol::
+This particular case case should be type-safe, but the type checker correctly stops the
```
Comment (alicederyn) [inline pep-9999.rst]:
```suggestion
This particular use case should be type-safe, but the type checker correctly stops the
```

File: pep-9999.rst (hunk 1 above)
Comment (dmoisset) [reply pep-9999.rst:38]: I see. As I mentioned, this is mostly "nit"; I think my wording is slightly more explicit but I'm fine either way

File: pep-9999.rst (hunk 2 above)
Comment (dmoisset) [reply pep-9999.rst]: 👍

File: pep-9999.rst (hunk 4 above)
Comment (dmoisset) [reply pep-9999.rst]: That's a simpler way to specify the constraint, I like it. Still I'd like the subject to be more explicit, e.g. "The runtime type of conforming values is not constrained to be a ``dict``"

File: pep-9999.rst (hunk 5 above)
Comment (dmoisset) [reply pep-9999.rst:145]: Not my fault that mine is better :P (still, this is a "nit")

**Review by dmoisset — COMMENTED** [review]
Added a couple of comments

File: pep-9999.rst (hunk 4 above)
Comment (alicederyn) [reply pep-9999.rst]:
```suggestion
1. The runtime type of a TypedMapping object is not constrained to be a ``dict``
```

**Pull request merged**